Automattic/wp-calypso
# Comments: experiment with routing middleware

This updates the comments navigation tab to dispatch a `ROUTE_TO` event. A new routing middleware then calls `page()` to update the current url.

### Testing Instructions
- Navigate to http://calypso.localhost:3000/comments/pending
- Select a site
- Click on a navigation tab, we should see a `ROUTE_TO` action being dispatched
- The tab should update as expected

![screen shot 2017-06-08 at 5 06 29 pm](https://user-images.githubusercontent.com/1270189/26955701-46dc251a-4c6d-11e7-8e1f-4aa7b445aff0.png)

- Also verify that `restore-last-location` behavior still works as expected.
- Visit some page other than root
- Then visit http://calypso.localhost:3000
- We should be redirected to that last page

cc @Automattic/lannister
